### PR TITLE
SDKS-1304: Cache revamp integrating events components

### PR DESCRIPTION
**Pull request by javrudsky**
# iOS SDK

## What did you accomplish?
- Integrating revamped events components
- Updated tests
- Created in memory database for testing purposes

### CODE REVIEW

File: Split/Storage/Impressions/ImpressionDao.swift [hunk 1]
```diff
@@ -86,6 +86,9 @@ class CoreDataImpressionDao: BaseCoreDataDao, ImpressionDao {
     }
 
     func delete(_ impressions: [Impression]) {
+        if impressions.count == 0 {
```
Comment (EmilianoSanchez) [inline Split/Storage/Impressions/ImpressionDao.swift:89]: shouldn't this check be done also on the EventsDao::delete method ?

File: Split/Storage/CoreDataHelper.swift [hunk 2]
```diff
@@ -72,12 +72,21 @@ class CoreDataHelper {
...
-            try persistentCoordinator.execute(deleteRequest, with: managedObjectContext)
+            let entities = try managedObjectContext.fetch(fetchRequest).compactMap { return $0 as? NSManagedObject }
+            for entity in entities {
+                managedObjectContext.delete(entity)
+            }
         } catch {
             Logger.e("Error while deleting \(entity.rawValue) entities from storage: \(error.localizedDescription)")
         }
+
+        //        let deleteRequest = NSBatchDeleteRequest(fetchRequest: fetchRequest)
```
Comment (EmilianoSanchez) [inline Split/Storage/CoreDataHelper.swift:84]: this comment could be removed if we are not keeping it for future references

File: Split/Storage/SplitDatabase.swift [hunk 3]
```diff
@@ -63,57 +62,14 @@ class CoreDataSplitDatabase: SplitDatabase {
     private let kDataModelName = "split_cache"
     private let kDataModelExtentsion = "momd"
     private let kDatabaseExtension = "sqlite"
```
Comment (EmilianoSanchez) [inline Split/Storage/SplitDatabase.swift:64]: Since these constants were moved to `CoreDataHelperBuilder`, check if they can be removed here.

**Review by EmilianoSanchez — COMMENTED** [review]
please Javi, check a few minor comments

File: Split/Storage/Impressions/ImpressionDao.swift (hunk 1 above)
Comment (javrudsky) [reply Split/Storage/Impressions/ImpressionDao.swift:89]: Yup. Added doc. Thank you!

File: Split/Storage/CoreDataHelper.swift (hunk 2 above)
Comment (javrudsky) [reply Split/Storage/CoreDataHelper.swift]: Done!

File: Split/Storage/SplitDatabase.swift (hunk 3 above)
Comment (javrudsky) [reply Split/Storage/SplitDatabase.swift]: Yeah, done doc!

**Review by EmilianoSanchez — APPROVED** [review]

**Pull request merged**